crop/structure.py

```diff
@@ -143,19 +175,125 @@ class ReadingsTinyTagsClass(BASE):
+    """
+    Class for reading the raw tiny tag data
+    """
+
+    __tablename__ = TINYTAGS_READINGS_TABLE_NAME
+
+    # columns
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    sensor_id = Column(
+        Integer,
+        ForeignKey("{}.{}".format(SENSOR_TABLE_NAME, ID_COL_NAME)),
+        nullable=False,
+    )
+
+    time_stamp = Column(DateTime, nullable=False)
+    temperature = Column(Integer, nullable=False)
+
+    time_created = Column(DateTime(), server_default=func.now())
+    time_updated = Column(DateTime(), onupdate=func.now())
+
+
+class ReadingsEnergyClass(BASE):
+    """
+    Class for reading the energy data
+    (monthly)
+    """
+
+    __tablename__ = ENERGY_READINGS_TABLE_NAME
+
+    # columns
+    id = Column(Integer, primary_key=True, autoincrement=True)
+
+    meter_point = Column(String, nullable=False)
+    period = Column(DateTime, nullable=False)
+    energy = Column(Float, nullable=False)
+
+    time_created = Column(DateTime(), server_default=func.now())
+    time_updated = Column(DateTime(), onupdate=func.now())
+
+class CropDataClass(BASE):
+    """
+    Class for reading the crop data
+    (from google sheets)
+    """
+
+    __tablename__ = CROP_SHEET_TABLE_NAME
+
+    # columns
+    id = Column(Integer, primary_key=True, autoincrement=True)
+
+    crop = Column(String, nullable=False)
+    propagation_date = Column(DateTime, nullable=False)
+    column = Column(Integer, nullable=False)
+    aisle = Column(String, nullable=False)
+    shelf = Column(Integer, nullable=False)
+    trays = Column(Integer, nullable=False)
+    m2 = Column(Integer, nullable=False)
+    supplier = Column(String, nullable=False)
+    batch_no = Column(String, nullable=False)
+    date_underlight = Column(DateTime, nullable=False)
+    week_harvested = Column(DateTime, nullable=False)
+    harvest_date = Column(DateTime, nullable=False)
+    traceability = Column(String, nullable=False)
+    surplus_waste_trays = Column(Integer, nullable=False)
+    est_disease_trays = Column(Integer, nullable=False)
+    mass_harvested = Column(Integer, nullable=False)
+    total_waste_p = Column(DECIMAL(5, 2), nullable=False) #percentages (?)
+    surplus_waste_p = Column(DECIMAL(5, 2), nullable=False)
+    total_waste_m2 = Column (String, nullable=False)
+    total_waste_p = Column(DECIMAL(5,2), nullable=False)
+    waste_explanation = Column(String, nullable=False) #no ref on what that is
+    yield_m2 = Column(Integer, nullable=False)
+    propagation_days = Column(Integer, nullable=False)
+    days_under_lights = Column(Integer, nullable=False)
+    unique_code = Column(String, nullable=False)
+    surplus_wasted = Column(Integer, nullable=False)
+    x_g = Column(Integer, nullable=False)
+    total_waste_g = Column(Integer, nullable=False)
+    projected_yeild_m = Column(Integer, nullable=False)
+    projected_yield_tot = Column(Integer, nullable=False)
+    num_trays_wasted = Column(String, nullable=True)
```

Comment:
If it is a number, perhaps it should be Float?

Reply:
Now that i have checked, they seem to be using integers for all these columns.  eg. 2659.

Reply:
change to floats